src/test/unit/algorithms/SpatialPoolerTest.cpp

```diff
@@ -1475,8 +1475,8 @@ TEST(SpatialPoolerTest, testInitPermanence) {
   SpatialPooler sp;
   Real synPermConnected = 0.2f;
   Real synPermActiveInc = 0.05f;
-  sp.initialize(inputDim, columnDim);
-  sp.setSynPermConnected(synPermConnected);
+  sp.initialize(inputDim, columnDim, 16u, 0.5f, true, -1, 10u, 0u, 0.01f, 0.1f,
```

Comment:
ok. this has probably been the reason for the method, you now have to specify all the constructor params prior to the synPermConnected. (pity c does not allow named paramaters like python does) 
But I guess it's ok, as one does not need to change synPermConnected that ofthen. 

Another way would be to remove `synPermConnected` even from the constructor params/initialize? But that might be a too much of a stretch. One could then operate the connected threshold only with synPermActiveInc/Dec. Also what should the default synPermConnected be? (0.2, 0.5, ...)

Reply:
@ctrl-z-9000-times thoughts?

Reply:
C++ already has a default value (IIRC 0.5f) and most people don't need to change it.  And if they do want non-default values then chances are: they want use all of the parameters anyways.  Its ugly but users probably won't need to copy-paste the default parameters like I did here.